justfile

```diff
@@ -0,0 +1,57 @@
+interface := "en0"
+ip := "127.0.0.1"
+repo_root := justfile_directory()
+scratch := justfile_directory() + "/.scratch"
+ipxe_result_link := scratch + "/ipxe-result"
+
+
+run: make-pxe-efi make-cert make-samples
+    go run . \
+        -interface "{{interface}}" \
+        -netboot-dir "{{scratch}}/samples" \
+        -tls-cert-file "{{scratch}}/tls.crt" \
+        -tls-key-file "{{scratch}}/tls.key" \
+        -dhcpv6-listen-port 20547 \
+        -http-listen-addr "{{ip}}:20080" \
+        -https-listen-addr "{{ip}}:20443" \
+        -tftp-listen-addr "{{ip}}:20069"
+
+make-pxe-efi: make-scratch
+    #!/bin/sh
+    if [ ! -e "{{ipxe_result_link}}/ipxe.efi" ]; then
+        nix build .#iPXE --out-link "{{ipxe_result_link}}"
+    else
+        echo "ipxe.efi: not re-building since it already exists in .scratch"
+    fi
```

Comment:
uh, just saving an eval here or?

Reply:
yeah, mostly just trying to get enough stuff to exist for it to plausibly work and test its interfaces... I found myself running `just` on every change, and the contents of the ipxe.efi was immaterial

Reply:
hm, I'd like to think nix would be smart enough to know that the text it read in for a build matches it's eval cache as well. I don't really understand why the re-eval in this path shouldn't be so fast as to be trivial.

anyway, it's fine.